lua/luasnip/session/snippet_collection.lua

```diff
@@ -59,16 +68,26 @@ local function insert_sorted_unique(t, k)
 	t[i] = k
 end
 
-local sort_mt = {
+local by_prio_snippets_mt = {
+	__index = function(s, k)
+		-- make new tables as they are indexed
+		return auto_creating_tables(s, k, 3)
+	end,
 	__newindex = function(t, k, v)
 		-- update priority-order as well.
 		insert_sorted_unique(t.order, k)
 		rawset(t, k, v)
 	end,
 }
+local prio_mt2 = {
```

Comment:
A comment for usage of this table? Since it's only used much later (declaring it here is good though :+1:)

Reply:
Nice!